src/parser/jobs/rdm/modules/MeleeCombos.js

```diff
@@ -0,0 +1,562 @@
+	finisher: ACTIONS.VERFLARE,
+}
+const DELAY_ACCELERATION_AVIALABLE_THRESHOLD = 4 // 4 seconds for 2 GCDs minus a 1 second window to activate before finisher
+
+export default class MeleeCombos extends Module {
+	static handle = 'meleecombos'
+	// Disable unused dependency detection because module requires gauge for the fabricated event but no other data that triggers the used dependency check
+	/* eslint-disable xivanalysis/no-unused-dependencies */
+	static dependencies = [
+		'combatants',
+		'suggestions',
+		'gauge',
+		'cooldowns',
+		'timeline',
+	]
+	/* eslint-enable xivanalysis/no-unused-dependencies */
+	static title = 'Melee Combos'
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {
+			by: 'player',
+			abilityId: ACTIONS.ACCELERATION.id,
+		}, this._onCast)
+		this.addHook('rdmCast', {by: 'player'}, this._onCast)
+		this.addHook('death', {to: 'player'}, this._onDeath)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_meleeCombos = {}
+
+	//Finisher Handling
+	_incorrectFinishers = {
+		verholy: 0,
+		verflare: 0,
+		delay: 0,
+	}
+
+	_startCombo(event) {
+		this._currentCombo = {
+			start: event.timestamp,
+			startmana: {
+				white: event.mana.white.beforecast,
+				black: event.mana.black.beforecast,
+			},
+			startprocs: {
+				verstone: this.combatants.selected.hasStatus(STATUSES.VERSTONE_READY.id),
+				verfire: this.combatants.selected.hasStatus(STATUSES.VERFIRE_READY.id),
+				acceleration: this.combatants.selected.hasStatus(STATUSES.ACCELERATION.id),
+			},
+			events: [event],
+			lastAction: event,
+		}
+	}
+
+	_breakComboIfExists() {
+		if (this._currentCombo) {
+			this._currentCombo.broken = true
+			this._endCombo()
+		}
+	}
+
+	_endCombo() {
+		this._meleeCombos[this._currentCombo.start] = this._currentCombo
+		delete this._currentCombo
+	}
+
+	_handleFinisher() {
+		const isFireReady = this.combatants.selected.hasStatus(STATUSES.VERFIRE_READY.id)
+		const isStoneReady = this.combatants.selected.hasStatus(STATUSES.VERSTONE_READY.id)
+		const combo = this._currentCombo
+		combo.finisher = {
+			recommendedActions: [],
+			recommendation: '',
+		}
+		const finisherUsed = combo.events[combo.events.length-1].ability
+
+		let recommendedFinisher = null
+		if (combo.startmana.white < combo.startmana.black) {
+			recommendedFinisher = this._outOfBalanceFinisher(combo.startmana.white, isStoneReady, WHITE_MANA_ACTIONS, combo.startmana.black, isFireReady, BLACK_MANA_ACTIONS)
+		} else if (combo.startmana.black < combo.startmana.white) {
+			recommendedFinisher = this._outOfBalanceFinisher(combo.startmana.black, isFireReady, BLACK_MANA_ACTIONS, combo.startmana.white, isStoneReady, WHITE_MANA_ACTIONS)
+		} else {
+			recommendedFinisher = this._inBalanceFinisher(combo.startmana.black, isFireReady, BLACK_MANA_ACTIONS, combo.startmana.white, isStoneReady, WHITE_MANA_ACTIONS)
+		}
+
+		if (recommendedFinisher instanceof Array) {
+			if (recommendedFinisher === FINISHERS) {
```

Comment:
seems like a peculiar way of representing it, but i'll water under the bridge it 'cus i don't gotta maintain it :blobmorning:

Reply:
The two alternatives I could come up with were to give special meaning to null (or a boolean value), or to return both finishers and give that special meaning... and since I wanted to display the sequence recommended for a delay (which meant an array anyway), returning both finishers seemed like the better alternative.